test/HarpoonKrakenTests.cpp

```diff
@@ -278,24 +323,77 @@ TEST_F(HarpoonKrakenTests, SendThreadSendHello) {
    uint8_t l = 'l';
    uint8_t o = 'o';
    uint8_t end = '\0';
-   const std::vector<uint8_t> expected{h,e,l,l,o,end};
-   
-   for (const auto c: expected){
+   const std::vector<uint8_t> expected{h, e, l, l, o, end};
+
+   for (const auto c : expected) {
       std::vector<uint8_t> p;
-      Harpoon::Battling res = client.Heave(p); 
-      EXPECT_EQ(res, Harpoon::Battling::CONTINUE); 
+      Harpoon::Battling res = client.Heave(p);
+      EXPECT_EQ(res, Harpoon::Battling::CONTINUE);
       ASSERT_EQ(p.size(), 1);
       EXPECT_EQ(p[0], c);
    }
-   
+
+   //Should now receive an empty chucnk to indicate end of stream:
+   Harpoon::Battling res = client.Heave(p);
+   EXPECT_EQ(res, Harpoon::Battling::VICTORIOUS) << "result: " << static_cast<int>(res) << ", VICTORIOUS: " << static_cast<int>(Harpoon::Battling::VICTORIOUS);
+
+   EXPECT_EQ(p.size(), 0);
+   done.wait();
+}
+
+TEST_F(HarpoonKrakenTests, ReceiveAbortDuringHello) {
+
+   const int port = GetTcpPort();
+   const std::string location = GetTcpLocation(port);
+   auto helloAbort = std::make_shared<HelloAbort>(location);
+   auto done = std::async(std::launch::async, &SendHelloExpectCancel, &helloAbort);
+
+   Harpoon client;
+   std::vector<uint8_t> p;
+   client.MaxWaitInMs(1000); // on purpose not the same timeout
+
+   Harpoon::Spear status = client.Aim(location);
+   EXPECT_EQ(status, Harpoon::Spear::IMPALED) << "1st: " << static_cast<int>(status) << ", IMPALED: " << static_cast<int>(Harpoon::Spear::IMPALED);
+
+   uint8_t h = 'h';
+   uint8_t e = 'e';
+   uint8_t l = 'l';
+   uint8_t o = 'o';
+   uint8_t end = '\0';
+
+   const std::vector<uint8_t> expected{h, e, l, l, o, end};
+   const size_t abortAt = 2;
+   size_t counter = 0;
+   for (const auto c : expected) {
+      std::vector<uint8_t> p;
+      auto res = Harpoon::Battling::CONTINUE;
+
+      if (abortAt == counter) {
+         LOG(INFO) << "Calling cancel at count:" << counter << std::endl;
```

Comment:
Still not sure I follow you,

The test calls "cancel" at a specific count and then the test verifies later by checking a flag that the receiving thread received the "cancel".

Please suggest with pseudo code how to make this more clear